### ISSUE TITLE
stairs that are not 45 degrees 

### ISSUE THREAD

**Issue by plankatron** [marian42/wavefunctioncollapse#19]
i can see why stairs would be 45 degrees, but do you have any ideas as to how to make stairs that are not ? 

typical staircase angle is 35 degrees

**Comment by Cons-Cat** [comment]
22.5 degree stairs would be interchangeable with 45 degree stairs, which might feel closer to what you want.

Perhaps you can change the way you reason about the world around a staircase to better suit a 35 degree staircase? If we need 45 degrees because tiles are 1x1x1 cubes, then to make 35 degree staircase tiles, simply change the height component.

35 / 45 ~= 0.778
1 * 0.778 = 0.778

So with 1x0.778x1 sized tiles, they're squashed enough to make 35 degree stairs work without any fundamental change to the WFC algorithm. I think this is the simplest solution.

**Comment by plankatron** [comment]
from staring at this project for a night now, it seems like there should be a way to create sets of stair modules that can only link together as a pair via the numbering constraints but i still dont fully understand the numbering scheme, anyways so it goes ..

**Comment by marian42** [comment]
Yeah the 45 degree angle was to allow the stairs to attach to the floor on both neighbor blocks. But some of the staris are already two blocks "deep", so they could be any angle. Usually it also works if one block reaches a bit into the next one. Making the stairs less steep is definitely on my list of things to do, along with some other scaling issues.
